aep/general/0140/aep.md.j2

```diff
@@ -1,5 +1,165 @@
+names **should** avoid including unnecessary words. In particular, avoid
+including adjectives that always apply and add little cognitive value. For
+example, a `proxy_settings` field might be as helpful as
+`shared_proxy_settings` if there is no unshared variant.
+
+**Important:** Field names often appear in generated client surfaces. Ensure
+they are appropriately descriptive and of suitable length.
+
+### Case
+
+{% tab proto %} Field definitions in protobuf files **must** use
+`lower_snake_case` names. These names are mapped to an appropriate naming
+convention in JSON and in generated code.
+
+{% tab oas %}
+
+Field definitions in OAS schemas **must** use `camelCase` names.
+
+{% endtabs %}
+
+**Note:** Examples throughout this AEP use `snake_case`. This should not be
+taken to mean that this guidance is protobuf-specific. Any guidance not in a
+protobuf tab applies to all APIs, with `snake_case` examples converted to
+`camelCase` for OAS APIs.
+
+Each word in a field name **must not** begin with a number, because it creates
+ambiguity when converting between snake case and camel case. Similarly, fields
+**must not** contain leading, trailing, or adjacent underscores.
+
+### Uniformity
+
+APIs **should** endeavor to use the same name for the same concept and
+different names for different concepts wherever possible. This includes names
+across multiple APIs, in particular if those APIs are likely to be used
+together.
+
+### Arrays / Repeated fields
+
+Arrays (in OAS) and repeated fields (in protobuf) **must** use the proper
+plural form, such as `books` or `authors`. On the other hand, singular fields
+**should** use the singular form such as `book` or `author`.
+
+### Prepositions
+
+Field names **should not** include prepositions (such as "with", "for", "at",
+"by", etc.). For example:
+
+- `error_reason` (**not** `reason_for_error`)
+- `author` (**not** `written_by`)
+
+It is easier for field names to match more often when following this
+convention. Additionally, prepositions in field names may also indicate a
+design concern, such as an overly-restrictive field or a sub-optimal data type.
+This is particularly true regarding "with": a field named `book_with_publisher`
+likely indicates that the book resource may be improperly structured and worth
+redesigning.
+
+**Note:** The word "per" is an exception to this rule, particularly in two
+cases. Often "per" is part of a unit (e.g. "miles per hour"), in which case the
+preposition must be present to accurately convey the unit. Additionally, "per"
+is often appropriate in reporting scenarios (e.g. "nodes per instance" or
+"failures per hour").
+
+### Adjectives
+
+For uniformity, field names that contain both a noun and an adjective
+**should** place the adjective _before_ the noun. For example:
+
+- `collected_items` (**not** `items_collected`)
+- `imported_objects` (**not** `objects_imported`)
+
+### Verbs
+
+Field names **must not** be named to reflect an intent or action. They **must
```

Comment:
Formatting nit: Put the part starting with "Rather" in a separate block (maybe a Note block?)

Reply:
Not sure I agree with this one - I think it's a continuation of the same thought.  I put it in a new paragraph to err on the side of reviewer feedback, but I don't think it makes sense in a Note: block.